src/uint/shl.rs

```diff
@@ -118,10 +133,15 @@ impl<const LIMBS: usize> Uint<LIMBS> {
         (Uint::<LIMBS>::new(limbs), Limb(carry))
     }
 
+    /// Computes `self << 1` in constant-time.
+    pub(crate) const fn shl1(&self) -> Self {
+        self.overflowing_shl1().0
+    }
+
     /// Computes `self << 1` in constant-time, returning [`ConstChoice::TRUE`]
-    /// if the most significant bit was set, and [`ConstChoice::FALSE`] otherwise.
+    /// if the most significant bit was set (i.e. if overflow occurred), and [`ConstChoice::FALSE`] otherwise.
     #[inline(always)]
-    pub(crate) const fn shl1_with_carry(&self) -> (Self, ConstChoice) {
+    pub(crate) const fn overflowing_shl1(&self) -> (Self, ConstChoice) {
```

Comment:
These should arguably remain named `shl1_with_carry` as it's carry, not overflow

Reply:
Yes, I agree.

Reply:
I mean, they should keep the `carry` suffix.